packages/app-desktop/gui/NoteEditor/utils/contextMenu.ts

```diff
@@ -100,17 +103,28 @@ export function menuItems(dispatch: Function): ContextMenuItems {
 			onAction: async (options: ContextMenuOptions) => {
 				await openItemById(options.resourceId, dispatch);
 			},
-			isActive: (itemType: ContextMenuItemType) => itemType === ContextMenuItemType.Image || itemType === ContextMenuItemType.Resource,
+			isActive: (itemType: ContextMenuItemType, options: ContextMenuOptions) => itemType === ContextMenuItemType.Image || (itemType === ContextMenuItemType.Resource && options.resourceId),
 		},
 		saveAs: {
 			label: _('Save as...'),
 			onAction: async (options: ContextMenuOptions) => {
-				const { resourcePath, resource } = await resourceInfo(options);
-				const filePath = await bridge().showSaveDialog({
-					defaultPath: resource.filename ? resource.filename : resource.title,
-				});
+				let defaultPath = '';
+				let resourcePath = null;
+				let resourceData = null;
+				if (options.resourceFilename) { defaultPath = options.resourceFilename; }
+				if (options.resourceContent) {
+					resourceData = options.resourceContent;
+				} else {
+					const info = await resourceInfo(options);
+					resourcePath = info.resourcePath;
+					const resource = info.resource;
+					if (!defaultPath) { defaultPath = resource.filename ? resource.filename : resource.title; }
+				}
+				const filePath = await bridge().showSaveDialog({ defaultPath });
 				if (!filePath) return;
-				await fs.copy(resourcePath, filePath);
+				if (resourceData) {
+					await fs.outputFile(filePath, resourceData);
+				} else { await fs.copy(resourcePath, filePath); }
```

Comment:
It's not great to have all this business logic in the view, and of course you then need to duplicate it in copyPathToClipboard.

Instead please refactor resourceInfo() to provide the required information.

As resourceInfo() grows in complexity please add unit tests too.